### DISCUSSION TITLE
Hacktoberfest 2021 - Win a Tshirt by contributing!

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2538] [Evennia Official]
![hacktoberfest](https://hacktoberfest.digitalocean.com/_nuxt/img/logo-hacktoberfest-full.f42e3b1.svg)

Like every year, Evennia is taking part in the [Hacktoberfest](https://hacktoberfest.digitalocean.com/)! Basically, it means you can win a limited-edition T-shirt by making 4 PRs to open-source projects in the period Oct 1-31 (you can also get a tree planted in your name if you prefer). To participate: 

1. [Sign up to Hacktoberfest](https://hacktoberfest.digitalocean.com/register). You just enter your normal github account for this. I think you can also sign up later and have previous PRs done from Oct 1-31 count.
2. Look over the Evennia [issue list](https://github.com/evennia/evennia/issues) and pick an issue to work on. If you are unsure about something, ask for clarification in the issue. You can pick any issue but below are some simpler ones to start with:
    - [Low-hanging fruit](https://github.com/evennia/evennia/issues?q=is%3Aissue+is%3Aopen+label%3AHacktoberfest%2C%22good+first+issue%22%2C%22help+needed%22). These vary in complexity, but most you should be able to do with pretty basic or minor Evennia knowledge. You should usually do these against the `develop` branch of Evennia (see the branch label).
    - [Make new unit tests](https://github.com/evennia/evennia/issues/1458). We always need more unit tests! This is also a great way to learn Evennia. Tests should always be done against `develop` branch.
    - [Documentation issues](https://github.com/evennia/evennia/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation): These require no coding, the docs are written in Markdown. All doc fixes should usually be done against `develop` branch (v1.0-dev) documentation in `evennia/docs/source/`. You don't need to be able to build the docs locally - just submit the text changes and I will incorporate and release the changes.
3. Do your changes and make a PR against Evennia in github. [See here for help on how to do this](https://www.evennia.com/docs/1.0-dev/Coding/Version-Control.html?highlight=git#forking-evennia). 
4. Your PR does not need to be merged to count towards the T-shirt but it needs a positive review or a special label set on it from me. [Here are some quality guidelines](https://hacktoberfest.digitalocean.com/resources/qualitystandards).

Hacktoberfest is a great way to challenge yourself and get you into contributing. 
Everyone profits! Good luck!

**Comment by kovitikus** [comment]
Aren't these t-shirts limited in supply? Meaning first come, first serve?

**Reply by Griatch** [reply]
Yes, there is a max limit for how many get a T-shirt. I don't know what that is based on. Personally I think the T-shirt is just a nice excuse to contribute though.

**Reply by Henddher** [reply]
First 50K users (github & gitlab) with 4 accepted PRs.

https://hacktoberfest.digitalocean.com/resources/participation

**Comment by mrseeker** [comment]
Is there a picture of this "limited t-shirt"?

**Reply by kovitikus** [reply]
Not that I could find anywhere on the site, so I doubt it.

**Reply by Henddher** [reply]
In the past, it matched the site.

**Comment by Griatch** [comment]
It's a new design every year. It's generally based on the logo for the year. Here are some designs from previous years: https://duckduckgo.com/?t=lm&q=hacktoberfest+tshirt&iax=images&ia=images

**Comment by Griatch** [comment]
As an update, we are a little over halfway into Hacktoberfest and so far I have reviewed and merged a whopping **17** Pull requests from contributors, closing no less than **6** bug/feature-request tickets in the process! This is great! 

Well done so far everyone! Let's see some more contributions in the second half of the month!  Lots of low-hanging fruit for you to try your hand at still.

**Comment by Griatch** [comment]
Hacktoberfest is drawing to a close and I think I can sum it up now. 
In the end we landed on some **22** PRs being merged (plus 2 which are still open but valid). Of these, **6** issues resolved as part of Hacktoberfest. So many PRs actually resolved new things that people noticed needed fixing but to which there were no formal ticket. Great! 

This is the best Hacktoberfest outcome so far. Thanks for the help, great work everyone!
